integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/database/device/DlmsDeviceSteps.java

```diff
@@ -751,20 +813,128 @@ public void registerNewKeys(final Map<String, String> inputSettings) {
+      afterXsecondsTheEncryptedSecretTableInTheSecretManagementDatabaseShouldContainKeysForDevice(
+          final int maxSecondsToWait,
+          final String secretTypeString,
+          final String deviceIdentification,
+          final Map<String, String> inputSettings) {
+    final Runnable task =
+        () -> {
+          this.theEncryptedSecretTableInTheSecretManagementDatabaseShouldContainKeysForDevice(
+              secretTypeString, deviceIdentification, inputSettings);
+        };
+
+    Wait.until(task, maxSecondsToWait, 1000);
+  }
+
+  @Then(
+      "the encrypted_secret table in the secret management database should contain {string} keys for device {string}")
+  public void theEncryptedSecretTableInTheSecretManagementDatabaseShouldContainKeysForDevice(
+      final String secretTypeString,
+      final String deviceIdentification,
+      final Map<String, String> inputSettings) {
+
+    final SecretType secretType = this.getSecretTypeByKeyTypeInputName(secretTypeString);
+
+    for (final String keyName : inputSettings.keySet()) {
+      final String secretStatus = inputSettings.get(keyName);
+      final String dbEncryptedSecretValue = SecurityKey.valueOf(keyName).getDatabaseKey();
+      final List<DbEncryptedSecret> dbEncryptedSecret =
+          this.encryptedSecretRepository.findSecrets(
+              deviceIdentification, secretType, SecretStatus.valueOf(secretStatus));
+
+      assertThat(dbEncryptedSecret)
+          .withFailMessage(
+              "No dbEncryptedSecret for %s with status %s found", secretTypeString, secretStatus)
+          .isNotEmpty();
+      final List<String> actualEncodedSecrets =
+          dbEncryptedSecret.stream()
+              .map(DbEncryptedSecret::getEncodedSecret)
+              .collect(Collectors.toList());
+      assertThat(actualEncodedSecrets)
+          .withFailMessage(
+              "Wrong dbEncryptedSecret for %s with status %s expected %s to be contained in: %s",
+              secretTypeString, secretStatus, dbEncryptedSecretValue, actualEncodedSecrets)
+          .contains(dbEncryptedSecretValue);
+    }
+  }
+
+  @Then(
+      "the encrypted_secret table in the secret management database should contain an EXPIRED and an ACTIVE key for device {string} or exactly the opposite")
+  public void
+      theEncryptedSecretTableInTheSecretManagementDatabaseShouldContainKeysForDeviceOrExactlyTheOpposite(
+          final String deviceIdentification, final Map<String, String> inputSettings) {
+    // Depending on the order in which the two request are processed the results (status of the
+    // secret) are either as specified as the first or second value
+    int countResultKeysRequest1 = 0;
+    int countResultKeysRequest2 = 0;
+
+    for (final String secretTypeString : inputSettings.keySet()) {
+
+      final String[] keyInfo = inputSettings.get(secretTypeString).split(",");
+
+      final String keyNameExpired = keyInfo[0];
+      final String keyNameActive = keyInfo[1];
```

Comment:
- Indeed naming is confusing keyNameExpired --> keyName1; keyNameActive --> keyName2
- It is complex because you must check that either one of the situations exists checking four variables in each situation. In your suggestion you forget to check the values of the secrets.
- Good point. IllegalArgumentException is now thrown more or less than two requests are defined in the settings